python/lsst/sims/catUtils/utils/LightCurveGenerator.py

```diff
@@ -0,0 +1,496 @@
+        return chunk
+
+    def get_pointings(self, ra, dec, bandpass=('u', 'g', 'r', 'i', 'z', 'y'), expMJD=None):
+        """
+        Inputs
+        -------
+        ra is a tuple indicating the (min, max) values of RA in degrees.
+
+        dec is a tuple indicating the (min, max) values of Dec in degrees.
+
+        bandpass is a str (i.e. 'u', 'g', 'r', etc.) or an iterable indicating
+        which filter(s) you want the light curves in.  Defaults to all six
+        LSST bandpasses.
+
+        expMJD is an optional tuple indicating a (min, max) range in MJD.
+        Defaults to None, in which case, the light curves over the entire
+        10 year survey are returned.
+
+        Outputs
+        -------
+        A 2-D list of ObservationMetaData objects.  Each row is a list of
+        ObservationMetaDatas that point to the same patch of sky, sorted by MJD.
+        Pointings will not be sorted or grouped by filter.
+        """
+
+        print('parameters', ra, dec, bandpass, expMJD)
+        if isinstance(bandpass, str):
+            obs_list = self._generator.getObservationMetaData(fieldRA=ra,
+                                                              fieldDec=dec,
+                                                              telescopeFilter=bandpass,
+                                                              expMJD=expMJD,
+                                                              boundLength=1.75)
+        else:
+            obs_list = []
+            for bp in bandpass:
+                sub_list = self._generator.getObservationMetaData(fieldRA=ra,
+                                                                  fieldDec=dec,
+                                                                  telescopeFilter=bp,
+                                                                  expMJD=expMJD,
+                                                                  boundLength=1.75)
+                obs_list += sub_list
+
+        if len(obs_list) == 0:
+            print("No observations found matching your criterion")
+            return None
+
+        # Group the OpSim pointings so that all of the pointings centered on the same
+        # point in the sky are in a list together (this will allow us to generate the
+        # light curves one pointing at a time without having to query the database for
+        # the same results more than once.
+        tol = 1.0e-12
+
+        obs_groups = []  # a list of list of the indices of the ObservationMetaDatas
+                         # in obs_list.  All of the ObservationMetaData in
+                         # obs_list[i] will point to the same point on the sky.
+
+        mjd_groups = []  # a list of lists of the MJDs of the ObservationMetaDatas
+                         # so that they can be sorted into chronological order before
+                         # light curves are calculated
+
+        for iobs, obs in enumerate(obs_list):
+            group_dex = -1
+
```

Comment:
Unless this is done over a small patch of the sky, this would be very slow because of the nested loop. On the other hand, to do exactly what is being done right now (no support for dithers and overlaps), one could actually group by fieldIDs. This would be much faster. Even to do the full, computation, this may be a good first step (or replacing this by some tree).